### PR TITLE
add solution for modern art 3

### PR DESCRIPTION
**Pull request by nwatx**

### CODE REVIEW

File: solutions/usaco-1114.mdx [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+const int MAX_N = 301;
+
+int A[MAX_N], dp[MAX_N][MAX_N];
+
+int main() {
+	setIO();
+	
+	int n; cin >> n;
+
```
Comment (SansPapyrus683) [inline solutions/usaco-1114.mdx:48]:
```suggestion
```

File: solutions/usaco-1114.mdx [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	for(int i = 0; i < n; i++) cin >> A[i];
+
```
Comment (SansPapyrus683) [inline solutions/usaco-1114.mdx:50]:
```suggestion
```

File: solutions/usaco-1114.mdx [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+	for(int i = 0; i <= n; i++) {
```
Comment (SansPapyrus683) [inline solutions/usaco-1114.mdx:51]: spaces for control statements please?

File: solutions/usaco-1114.mdx [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+		for(int j = 0; j <= n; j++) {
+			dp[i][j] = MAX_N + 1;
```
Comment (SansPapyrus683) [inline solutions/usaco-1114.mdx:53]: wait hol up `MAX_N` is alr 301 why you making it 302
can you just make `MAX_N` 300 bc that's the actual max value for n

File: solutions/usaco-1114.mdx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+		}
+	}
+
+	for(int i = 0; i < MAX_N; i++) dp[i][i] = 1;
+
```
Comment (SansPapyrus683) [inline solutions/usaco-1114.mdx:58]:
```suggestion
```

File: solutions/usaco-1114.mdx [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+	for(int s = 1; s <= n; s++) {
+		for(int i = 0; i <= n - s; i++) {
+			for(int j = i; j < i + s; j++) {
+				int k = i + s - 1;
+				if(A[i] == A[k])
```
Comment (SansPapyrus683) [inline solutions/usaco-1114.mdx:63]: maybe add a comment so people know where the algo explained in the editorial comes in

File: solutions/usaco-1114.mdx [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+					dp[i][k] = min(dp[i][k], dp[i][j] + dp[j + 1][k] - 1);
+				dp[i][k] = min(dp[i][k], dp[i][j] + dp[j + 1][k]);
+			}
+		}
+	}
+
```
Comment (SansPapyrus683) [inline solutions/usaco-1114.mdx:69]:
```suggestion
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-1114.mdx [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+int A[MAX_N], dp[MAX_N][MAX_N];
+
+int main() {
+	setIO();
+	
+	int n; cin >> n;
+	for (int i = 0; i < n; i++) cin >> A[i];
+	for (int i = 0; i <= n; i++) {
+		for(int j = 0; j <= n; j++) {
```
Comment (nathangong) [inline solutions/usaco-1114.mdx:50]:
```suggestion
		for (int j = 0; j <= n; j++) {
```

File: solutions/usaco-1114.mdx [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+	}
+
+	for (int i = 0; i < MAX_N; i++) dp[i][i] = 1;
+
+	for (int s = 1; s <= n; s++) {
+		for (int i = 0; i <= n - s; i++) {
+			for (int j = i; j < i + s; j++) {
+				int k = i + s - 1;
+				if (A[i] == A[k])
+					dp[i][k] = min(dp[i][k], dp[i][j] + dp[j + 1][k] - 1);
```
Comment (SansPapyrus683) [inline solutions/usaco-1114.mdx:62]: does this work the line length?

File: solutions/usaco-1114.mdx [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+[Official Editorial](http://www.usaco.org/current/data/sol_prob2_gold_feb21.html)
+
+## Explanation
+
+Define $\texttt{dp}[i][j]$ as the minimum number of paintings to paint the range
+$[i, j]$. Then, one of two things can happen when combining ranges
+$[i, j], [j + 1][k]$:
+
+1. $A[i] == A[k]$. This means that the range $[i, k]$ has a common color for the
+   endpoint. Therefore, we can "save" a color when merging the two ranges.
```
Comment (nathangong) [inline solutions/usaco-1114.mdx:17]:
```suggestion
1. $A[i] == A[k]$. This means that the range $[i, k]$ has a common color for each
   endpoint. Therefore, we can "save" a color when merging the two ranges.
```

File: solutions/usaco-1114.mdx [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+   $$
+   \texttt{dp}[i][k] = \min(\texttt{dp}[i][j] + \texttt{dp}[j + 1][k] - 1)
```
Comment (nathangong) [inline solutions/usaco-1114.mdx:19]:
```suggestion
   \texttt{dp}[i][k] = \min(\texttt{dp}[i][k], \texttt{dp}[i][j] + \texttt{dp}[j + 1][k] - 1)
```

**Comment by nathangong** [comment]
Also for some reason the internal solution doesn't popup when you search for the problem and click on its solution in the Problems tab. You might have to change the type to internal in another file.

**Comment by maggie-j-liu** [comment]
> Also for some reason the internal solution doesn't popup when you search for the problem and click on its solution in the Problems tab. You might have to change the type to internal in another file.

I think this is a known issue, but if the website deploys then all the problems should be linked fine

**Comment by nathangong** [comment]
Ah ok nvm then

File: solutions/usaco-1114.mdx (hunk 9 above)
Comment (nathangong) [reply solutions/usaco-1114.mdx:62]: I think it's around 74 characters so it should be fine

**Review by nathangong — APPROVED** [review]

File: solutions/usaco-1114.mdx [hunk 12]
```diff
@@ -0,0 +1,73 @@
...
+	int n; cin >> n;
+	for (int i = 0; i < n; i++) cin >> A[i];
+	for (int i = 0; i <= n; i++) {
+		for (int j = 0; j <= n; j++) {
+			dp[i][j] = MAX_N + 1;
+		}
+	}
+
+	for (int i = 0; i < MAX_N; i++) dp[i][i] = 1;
+
```
Comment (SansPapyrus683) [inline solutions/usaco-1114.mdx:56]:
```suggestion
```
after this everything should be ok

Comment (SansPapyrus683) [reply solutions/usaco-1114.mdx:56]: also add comments?

**Review by SansPapyrus683 — APPROVED** [review]

**Review by ar492 — APPROVED** [review]

File: solutions/usaco-1114.mdx [hunk 13]
```diff
@@ -0,0 +1,72 @@
...
+int A[MAX_N], dp[MAX_N][MAX_N];
+
+int main() {
+	setIO();
+	
+	int n; cin >> n;
+	for (int i = 0; i < n; i++) cin >> A[i];
+	for (int i = 0; i <= n; i++) {
+		for (int j = 0; j <= n; j++) {
```
Comment (dongliu0426) [inline solutions/usaco-1114.mdx:50]:
```suggestion
	for (int i = 0; i < n; i++) {
		for (int j = 0; j < n; j++) {
```

File: solutions/usaco-1114.mdx [hunk 14]
```diff
@@ -0,0 +1,72 @@
...
+			dp[i][j] = MAX_N + 1;
+		}
+	}
+
+	for (int i = 0; i < MAX_N; i++) dp[i][i] = 1;
```
Comment (dongliu0426) [inline solutions/usaco-1114.mdx:55]: < n

**Review by dongliu0426 — CHANGES_REQUESTED** [review]

File: solutions/usaco-1114.mdx [hunk 15]
```diff
@@ -0,0 +1,72 @@
+---
+id: usaco-1114
+source: USACO Gold 2021 February
+title: Modern Art 3
+author: Neo Wang
+---
+
+[Official Editorial](http://www.usaco.org/current/data/sol_prob2_gold_feb21.html)
```
Comment (ryanchou-dev) [inline solutions/usaco-1114.mdx:8]:
```suggestion
[Official Editorial (C++)](http://www.usaco.org/current/data/sol_prob2_gold_feb21.html)
```

**Review by ryanchou-dev — APPROVED** [review]
nice editorial ❤️

File: solutions/usaco-1114.mdx (hunk 6 above)
Comment (SansPapyrus683) [reply solutions/usaco-1114.mdx]: wait bump

Comment (nwatx) [reply solutions/usaco-1114.mdx]: @SansPapyrus683 check now

Comment (SansPapyrus683) [reply solutions/usaco-1114.mdx]: bruh
i guess it's ok but

**Pull request merged**